.github/workflows/deploy.yml

```diff
@@ -0,0 +1,43 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch: { }
+
+jobs:
+  tests:
+    name: Tests
+    uses: ./.github/workflows/test.yml
+  deploy-snapshots:
+    name: Deploy snapshot artifacts
+    needs: [ tests ]
+    runs-on: ubuntu-latest
+    if: github.repository == 'camunda/zeebe'
```

Comment:
:question: Why the check? Forks wouldn't have the right secrets to do this anyway, right? Would it be more appropriate to check for the presence of the secrets for publishing instead? Or what about using environments? Can we check if a given environment exists? :thinking:

Reply:
> Would it be more appropriate to check for the presence of the secrets for publishing instead

Hm, maybe? I mainly added this as a courtesy for forks so they don't get failing pipelines on the main branch but tbh. that ship has sailed since forks will also not have access to our self-hosted runners.

Reply:
I haven't looked at environments in detail but my first impression is that they are a bit "heavy" and don't really match what we are doing - just uploading snapshot artifacts.